lib/cubit/graph_cubit.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:graphview/GraphView.dart';
+import 'package:onyx/cubit/page_cubit.dart';
+import 'package:onyx/store/page_store.dart';
+
+class GraphState {
+  final Graph graph;
+  final List<Node> recursionExist;
+  final Map<String, Node> titleNode;
+  final Map<Node, Map<PageModel, bool>> dataNode;
+
+  GraphState({required this.graph, required this.recursionExist, required this.titleNode, required this.dataNode});
+}
+
+class GraphCubit extends Cubit<GraphState> {
+  final PageCubit cubit;
+  final Graph graph = Graph()..isTree = false;
+  Map<String, Node> titleNode = {};
+  Map<Node, Map<PageModel, bool>> dataNode = {};
+  final List<Node> recursionExist = [];
+
+  Node? getNodeByUid(String uid) => state.titleNode[uid];
+
+  Map<PageModel, bool>? getPageModelForNode(Node node) => state.dataNode[node];
+
+  Map<PageModel, bool>? getPageModelByTitle(String uid) {
+    final node = getNodeByUid(uid);
+    if (node != null) {
+      return getPageModelForNode(node);
+    }
+    return null;
+  }
+
+  GraphCubit(this.cubit) : super(GraphState(graph: Graph(), recursionExist: [], titleNode: {}, dataNode: {})) {
+    init();
+  }
+
+  void createGraphNodes(allPages){
+     for (final entry in allPages) {
+      final PageModel page = entry['page'] as PageModel;
+      final bool isJournal = entry['isJournal'] as bool;
+
+      if (page.uid.isNotEmpty && page.title.isNotEmpty) {
+        late final Node node;
+        if (isJournal && page.uid.contains('/')) {
+          final numericDate = int.parse(page.uid.replaceAll('/', ''));
+          node = Node.Id(numericDate);
+          titleNode[page.uid] = node;
+          graph.addNode(node);
+          dataNode[node] = {page: isJournal};
+        } else if (!isJournal) {
+          node = Node.Id(page.uid.hashCode);
+          titleNode[page.uid] = node;
+          graph.addNode(node);
+          dataNode[node] = {page: isJournal};
+        }
+       
+      }
+    }
+  }  
+
+  void createGraphEdges(allPages){
```

Comment:
```suggestion
  void createGraphEdges(SomeType allPages){
```
This and above should be typed